mailbox/elasticsearch/src/test/java/org/apache/james/mailbox/elasticsearch/events/ElasticSearchListeningMessageSearchIndexTest.java

```diff
@@ -442,4 +445,65 @@ void deleteAllShouldNotThrowWhenEmptyIndex() {
             .doesNotThrowAnyException();
     }
 
+    @Test
+    void retrieveIndexedFlagsShouldRetrieveFlags() {
+        Flags flags = new Flags();
+        flags.add(Flags.Flag.RECENT);
+        flags.add(Flags.Flag.DRAFT);
+
+        SimpleMailboxMessage message = MESSAGE_BUILDER.messageId(MESSAGE_ID_4)
+            .uid(MESSAGE_UID_4)
+            .flags(flags)
+            .build();
+
+        testee.add(session, mailbox, message).block();
+        elasticSearch.awaitForElasticSearch();
```

Comment:
Simple question: can you confirm if these awaits are needed?

Reply:
Seems not really but maybe safer to flush the indices, before trying to do a get operation on it?

Reply:
This depends, documents don't need to be indexed to be get, I wonder if the flush is needed.

Reply:
Seems working without so let's try it out